singer_sdk/streams/rest.py

```diff
@@ -244,7 +244,7 @@ def _request(
         if self._LOG_REQUEST_METRICS:
             extra_tags = {}
             if self._LOG_REQUEST_METRIC_URLS:
-                extra_tags["url"] = prepared_request.path_url
+                extra_tags["url"] = prepared_request.url
```

Comment:
This includes any URL query params, so it could lead to secrets leaking into metrics logs.

Wdyt @aaronsteers?

Reply:
I wonder if we should just add `RESTStream.url_path_as_sensitive` and `url_params_as_sensitive` and maybe even `url_root_as_sensitive`.

@visch - in our conversation earlier I misunderstood and was thinking that you only wanted to include the base url.

If we want to include arga as well (and I definitely think it's valuable to do so!), then maybe we just ask devs to tell us what can/can't be allowed in logs.

Reply:
Also - If we put the args here as well in url, then there might be value in a separate entry for `path` or `endpoint` that excludes them...

Reply:
@aaronsteers I don't see the URL params as generally useful to include _as a metric tag_, since the main purpose of metrics is not to debug the tap operation, but rather monitor it and analyze its performance.

Maybe in a more generic "The request failed, and these were the request details"-type of debug log message, it'd be useful to include the URL params, but they'd hardly bring any valuable info to annotate the metrics timeseries as is being done here.

Reply:
Need a way to debug that's easier than today, I don't really care if it's in a metric or not 🤷

Reply:
@edgarrmondragon and @visch - Yeah, I think we're going to be blocked generically from giving robust debug info unless we add something like the sensitivity flags I suggested above.

> I wonder if we should just add
> 
> * `RESTStream.url_path_as_sensitive` and
> * `RESTStream.url_params_as_sensitive` and maybe even
> * `RESTStream.url_root_as_sensitive`.

I'd opt to keep the metrics fairly clean, perhaps adding the base url so it isn't just a partial path.

And I'd probably also opt to taking a pass through the error log and, upon failure, print _as much detail as is not marked sensitive_.

Reply:
I thought `_LOG_REQUEST_METRIC_URLS` was a sensitivity flag  https://github.com/meltano/sdk/blob/3e819ddccf17eb2431050363caef29966112fdfe/singer_sdk/streams/rest.py#L57

Reply:
@visch and @aaronsteers So, looking into this, it seems that `PreparedRequest.path_url` already contains URL params and may leak secrets 🤦‍♂️:

```python
>>> import requests
>>> s = requests.Session()
>>> req = requests.Request(method="GET", url="https://httbin.org/test", params={"secret": "abc"})
>>> prep = s.prepare_request(req)
>>> prep.path_url
'/test?secret=abc'
```

We need to get rid of this from the metric log entirely, or at least parse the url and pass only the path here:

```
from urllib.parse import urlparse

extra_tags["url"] = urlparse(prepared_request.url).path
```

That said, I still think that there should be a dedicated for request details elsewhere.